fw/http_msg.c

```diff
@@ -555,10 +555,12 @@ tfw_http_msg_hdr_close(TfwHttpMsg *hm)
 		 * RFC 7230 3.2.2: all duplicates of special singular
 		 * headers must be blocked as early as possible,
 		 * just when parser reads them.
+		 *
+		 * RFC 9113 8.3: Pseudo-header fields MUST not appear
+		 * in a trailer section.
 		 */
 		if (id < TFW_HTTP_HDR_NONSINGULAR) {
-			if (WARN_ON_ONCE(!TFW_MSG_H2(hm)
-					 || id != TFW_HTTP_HDR_COOKIE))
+			if (!TFW_MSG_H2(hm) || id != TFW_HTTP_HDR_COOKIE)
```

Comment:
Well, AFAIK even if the header type would be `Literal Header Field without Indexing`, e.g. name and value would be passed as pure strings, we would still end up here and effectively filter out pseudo-headers in trailers, if any, so it would work for not fully-indexed headers as well.